src/Sparrow/Utils/SafeFileStream.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.IO;
+using Microsoft.Win32.SafeHandles;
+
+namespace Sparrow.Utils
+{
+    public static class SafeFileStream
+    {
+        private static void AvoidKillingTheProcessWhenOutOfMemory(FileStream file)
```

Comment:
Out of _disk space_

Reply:
https://github.com/ravendb/ravendb/pull/5106